src/trading/TradingEscapeHatch.sol

```diff
@@ -0,0 +1,94 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/trading/ITradingEscapeHatch.sol';
+import 'ROOT/trading/ICash.sol';
+import 'ROOT/trading/IOrders.sol';
+import 'ROOT/trading/IShareToken.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+
+contract TradingEscapeHatch is DelegationTarget, ITradingEscapeHatch {
+    using SafeMathUint256 for uint256;
+
+    //      market             outcome  value
+    mapping(address => mapping(uint8 => uint256)) private frozenShareValues;
+
+    function claimSharesInUpdate(IMarket _market) public onlyInBadTimes returns(bool) {
+        require(address(_market) != address(0));
+
+        uint8 _numOutcomes = _market.getNumberOfOutcomes();
+        ICash _marketCurrency = _market.getDenominationToken();
+
+        for (uint8 _outcome = 0; _outcome < _numOutcomes; ++_outcome) {
+            IShareToken _shareToken = _market.getShareToken(_outcome);
+            uint256 _sharesOwned = _shareToken.balanceOf(msg.sender);
+            if (_sharesOwned > 0) {
+                uint256 _frozenShareValue = getFrozenShareValue(_market, _numOutcomes, _outcome);
+                _shareToken.destroyShares(msg.sender, _sharesOwned);
+                uint256 _amountToTransfer = _sharesOwned.fxpMul(_frozenShareValue, 1 ether);
+                require(_marketCurrency.transferFrom(_market, msg.sender, _amountToTransfer));
+            }
+        }
+        return true;
+    }
+
+    function getFrozenShareValue(IMarket _market, uint8 _numOutcomes, uint8 _outcome) internal returns(uint256) {
+        require(0 <= _outcome && _outcome < _numOutcomes);
```

Comment:
This can be simplified to `require(outcome < _numOutcomes)`.  We needed the `0` check before because Serpent didn't have types so this could have been negative.